src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -419,9 +421,6 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
                 return null;
             }
 
-            // If id_token is received using code only flow, no need to validate chash.
-            await ValidateOpenIdConnectProtocolAsync(jwt, message, false);
```

Comment:
The original Response message has only a code. That code is redeemed for a token + id_token inside the OIDCMiddleware. I don't know why we would pass the original message + id_token from the tokenendpoint to the validator.